lgc/elfLinker/FetchShader.cpp

```diff
@@ -114,42 +134,74 @@ Module *FetchShader::generate() {
       while (!func.use_empty()) {
         auto call = cast<CallInst>(func.use_begin()->getUser());
         Value *replacement = nullptr;
-        switch (cast<ConstantInt>(call->getArgOperand(0))->getZExtValue()) {
-        case static_cast<unsigned>(UserDataMapping::VertexBufferTable): {
-          // Need to extend 32-bit vertex buffer table address to 64 bits.
-          AddressExtender extender(fetchFunc);
-          Value *highAddr = call->getArgOperand(1);
-          builder.SetInsertPoint(&*fetchFunc->front().getFirstInsertionPt());
-          replacement = extender.extend(fetchFunc->getArg(m_vsEntryRegInfo.vertexBufferTable), highAddr,
-                                        call->getType(), builder);
-          break;
-        }
-        case static_cast<unsigned>(UserDataMapping::BaseVertex):
-          replacement = fetchFunc->getArg(m_vsEntryRegInfo.baseVertex);
-          break;
-        case static_cast<unsigned>(UserDataMapping::BaseInstance):
-          replacement = fetchFunc->getArg(m_vsEntryRegInfo.baseInstance);
-          break;
-        case static_cast<unsigned>(ShaderInput::VertexId):
-          builder.SetInsertPoint(&*fetchFunc->front().getFirstInsertionPt());
-          replacement = builder.CreateBitCast(fetchFunc->getArg(m_vsEntryRegInfo.sgprCount + m_vsEntryRegInfo.vertexId),
-                                              builder.getInt32Ty());
-          break;
-        case static_cast<unsigned>(ShaderInput::InstanceId):
-          builder.SetInsertPoint(&*fetchFunc->front().getFirstInsertionPt());
-          replacement = builder.CreateBitCast(
-              fetchFunc->getArg(m_vsEntryRegInfo.sgprCount + m_vsEntryRegInfo.instanceId), builder.getInt32Ty());
-          break;
-        default:
-          llvm_unreachable("Unexpected special user data or shader input");
-        }
+        replacement = getReplacementForInputBuiltIn(call);
         call->replaceAllUsesWith(replacement);
         call->eraseFromParent();
       }
     }
   }
+}
 
-  return fetchFunc->getParent();
+// =====================================================================================================================
```

Comment:
Add function description comments.

Reply:
I'll do it, but I have a problem with how much extra work our coding standards add for no extra value.  What extra information will the function description add that is not covered by the "@return".  If I add the description, but not the "@return", then I get a comment that the code standards document says I have to have the "@return".  On top of that, these comments are often just repeating the same information that is in the function name.

Reply:
I think AMD follows this style in other projects:
```
// =====
// Function descriptions
//
// NOTE: Extra comments that need special attention.
//
// @param X :
// @param Y :
// @return ...
```